octree/include/pcl/octree/octree_nodes.h

```diff
@@ -225,7 +223,12 @@ class OctreeBranchNode : public OctreeNode {
   inline OctreeNode*&
   operator[](unsigned char child_idx_arg)
   {
-    assert(child_idx_arg < 8);
+    if (child_idx_arg >= 8) {
+      PCL_ERROR(
+          "[pcl::octree::OctreeBranchNode::operator[]] Child index must be < 8!\n");
+      OctreeNode* node = nullptr;
+      return node;
```

Comment:
Ubuntu 18.04 `error: reference to local variable 'node' returned [-Werror=return-local-addr]`, need to redesign the API?

Reply:
Would also keep assert here, add a note to the param documentation that it has to be less than 8, if you like

Reply:
Btw this module uses lots of raw pointer, which differs from other modules. Would it be recommendable to refactor this module to use reference or smart pointer?

Reply:
That would probably break a lot of API, right? I think the benefits would be too small to justify a refactoring

Reply:
Yes... But codes like `container = ***leaf_container_vector_it_arg;` in `octree_base.hpp` are just too hard to read.

Reply:
We could add steps in between like `LeafContainerT* some_name_here = **leaf_container_vector_it_arg; container = *some_name_here;` to make it clearer what happens (the compiler should be able to optimize that out)

Reply:
Ok, maybe I'll create another PR for this.